modules/geolocation/geolocation.py

```diff
@@ -129,3 +140,210 @@ def calculate_pixel_to_geo_mapping(self):
+            Array containing components of the o vector
+        """
+        # get GPS module coordinates
+        gpsModule = np.zeros(3)
+        gpsModule[0] = origin[0] + latitude
+        gpsModule[1] = origin[1] + longitude
+        gpsModule[2] = origin[2] + altitude
+
+        # transform gps-to-camera offset from plane space to world space by applying inverse
+        # rotation matrix using (AB)^-1 = B^-1 A^-1
+        rotationMatrix = np.matmul(planeRotation, cameraRotation)
+        transposedMatrix = np.transpose(rotationMatrix)
+        rotatedOffset = transposedMatrix.dot(gpsCameraOffset)
+
+        # get camera coordinates in world space
+        oVector = np.add(rotatedOffset, gpsModule)
+
+        return oVector
+
+    def __calculate_c_vector(self, compoundRotationMatrix: np.ndarray, cVectorCameraSpace=np.array([1, 0, 0])) -> np.ndarray:
+        """
+        Returns a numpy array that contains the components of the c vector
+
+        Parameters
+        ----------
+        compoundRotationMatrix: numpy array
+            Array containing rotation matrix for camera and plane rotations
+        cVectorCameraSpace: numpy array
+            Array containing cVector in camera space (default value assumes cVector points along the roll axis)
+
+        Returns
+        -------
+        cVector:
+            Array containing components of the c vector
+        """
+        # apply rotation matrix to camera direction
+        # note: this assumes that camera euler angles are w.r.t. plane space
+        cVector = compoundRotationMatrix.dot(cVectorCameraSpace)
+
+        # normalize since magnitude doesn't matter
+        norm = np.linalg.norm(cVector)
+        if(norm != 0) :
+            cVector = cVector / norm
+
+        cVector = np.squeeze(cVector)
+        return cVector
+
+    def __calculate_u_vector(self, compoundRotationMatrix: np.ndarray, uVectorCameraSpace=np.array([0, 1, 0])) -> np.ndarray:
+        """
+        Returns a numpy array that contains the components of the u vector (one of the camera rotation vectors)
+
+        Parameters
+        ----------
+        compoundRotationMatrix: numpy array
+            Array containing rotation matrix for camera and plane rotations
+        uVectorCameraSpace: numpy array
+            Array containing uVector in camera space (default value assumes uVector points along the pitch axis)
+            
+        Returns
+        -------
+        uVector: numpy array
+            Array containing camera rotation vector
+        """
+        # apply camera rotation to camera direction
+        # note: this assumes that camera euler angles are w.r.t. plane space
+        uVector = compoundRotationMatrix.dot(uVectorCameraSpace)
+
+        # normalize output since camera direction magnitude doesn't matter
+        norm = np.linalg.norm(uVector)
+        if (norm != 0) :
+            uVector = uVector / norm
+
+        uVector = np.squeeze(uVector)
+        return uVector
```

Comment:
This should be multiplied by a field-of-view constant so that the ratio between this and the **c** vector are correct.